src/abstractdataframe/sort.jl

```diff
@@ -73,14 +73,14 @@ DFPerm(o::O, df::DF) where {O<:Ordering, DF<:AbstractDataFrame} = DFPerm{O,DF}(o
 col_ordering(o::DFPerm{O}, i::Int) where {O<:Ordering} = o.ord
 col_ordering(o::DFPerm{V}, i::Int) where {V<:AbstractVector} = o.ord[i]
 
-Base.@propagate_inbounds Base.getindex(o::DFPerm, i::Int, j::Int) = o.df[i, j]
-Base.@propagate_inbounds Base.getindex(o::DFPerm, a::DataFrameRow, j::Int) = a[j]
+Base.@propagate_inbounds Base.getindex(o::DFPerm, i::Int, j::Union{Int,Symbol}) = o.df[i, j]
+Base.@propagate_inbounds Base.getindex(o::DFPerm, a::DataFrameRow, j::Union{Int,Symbol}) = a[j]
 
 function Sort.lt(o::DFPerm, a, b)
-    @inbounds for i = 1:ncol(o.df)
+    @inbounds for (i,n) = enumerate(names(o.df))
         ord = col_ordering(o, i)
-        va = o[a, i]
```

Comment:
Just curious: why doesn't `i` work here?

Reply:
This is why I say that it is messy.

The reason is line:
```
issorted(eachrow(df), ordering(df, cols_new, lt, by, rev, order))
```
which calls `ordering` which calls `DFPerm` like
```
DFPerm(ords, df[newcols])
```

So we see that in `lt` value of `o.df` is a data frame with `newcols` but `a` and `b` are data frames with all columns from `df` because they are produced by `eachrow(df)`.

However, given your comment I have decided to start cleaning it up a bit and simplified the code (now it does not use `eachrow`).